Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -49,6 +49,19 @@ public class ByteBuffer {
         return self
     }
 
+    public func put(_ value: ByteBuffer) {
+        array.append(contentsOf: value.array)
+    }
+
+    public func put(_ value: ByteBuffer, offset: UInt = 0, maxBytes: UInt?) {
```

Comment:
* if we default offset to 0, then it should come after variables that do not have defaults
* investigate combining both put functions to single function.

Reply:
+1, these are the same function if `offset` and `maxBytes` have defaults

Reply:
merging into one function